frontend/pages/[resourceType].tsx

```diff
@@ -0,0 +1,107 @@
+import { GetServerSideProps, InferGetServerSidePropsType } from 'next';
+import { Button, Grid, Divider } from '@mantine/core';
+import Link from 'next/link';
+
+/**
+ * Component which displays list of all resources of some type as passed in by (serverside) props
+ * @returns component with list of resource (by id) buttons that are links to that resource's details
+ */
+export default function ResourceList({ ids, resourceType }: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  const idItems = ids.map(id => {
+    return (
+      <Link href={`/${resourceType}/${id}`} key={id}>
+        <Button
+          fullWidth
+          color="cyan"
+          radius="md"
+          size="md"
+          variant="subtle"
+          styles={() => ({
+            root: {
+              padding: '2px'
+            },
+            inner: {
+              justifyContent: 'left'
+            }
+          })}
+        >
+          <div>
+            {resourceType}/{id}
+          </div>
+        </Button>
+      </Link>
+    );
+  });
+  return (
+    <div
+      style={{
+        width: '78vw'
+      }}
+    >
+      <Grid columns={7}>
+        <Grid.Col offset={3} span={2} style={{ paddingTop: '5px' }}>
+          <h2 style={{ color: 'gray', marginTop: '0px', marginBottom: '8px' }}>{`${resourceType} IDs`}</h2>
+        </Grid.Col>
+        <Grid.Col
+          span={2}
+          style={{
+            paddingTop: '5px'
+          }}
+        ></Grid.Col>
+      </Grid>
+      <Divider my="md" style={{ marginTop: '14px' }} />
+      <div>
+        <div
+          style={{
+            textAlign: 'left',
+            overflowWrap: 'break-word',
+            padding: '10px',
+            backgroundColor: '#FFFFFF',
+            border: '1px solid',
+            borderColor: '#DEE2E6',
+            borderRadius: '20px',
+            marginTop: '10px',
+            marginBottom: '20px',
+            marginLeft: '150px',
+            marginRight: '150px'
+          }}
+        >
+          {idItems.length > 0 ? ( //if items exist
+            <ul>{idItems}</ul>
+          ) : (
+            <text>
+              No <i>{`${resourceType}`}</i> resources available
+            </text>
+          )}
+        </div>
+      </div>
+    </div>
+  );
+}
+
+/**
+ * Serverside props pulls id data of a certain resourceType to pass to the page before it's sent to browser
+ * @returns props for the [resourceType] page that pass resourceType and ids of resources of that type
+ */
+export const getServerSideProps: GetServerSideProps<{
+  ids: (string | undefined)[];
+  resourceType: string;
```

Comment:
Would it be worth it to change resourceType from type string to something more specific like `'Measure' | 'Library'`?

Reply:
I def considered that. It's a bit of a tossup, but I figured in case we wanted to have MeasureReports in here some day I might leave it more flexible for now. *shrug* could go either way lol

Reply:
I created a [util file](https://github.com/projecttacoma/measure-repository/blob/main/frontend/util/types/fhir.ts) when implementing the resource JSON rendering, so we could add to that if we really wanted to and create a custom type for just Measure and Library strings. But I think it's fine to stay flexible for now. I don't anticipate us needing to constrain `resourceType` to certain values since it's assumed that the server will already restrict what resource types can exist on the server

Reply:
`export type ArtifactResourceType = FhirArtifact["resourceType"];` would be a nice dynamic solution